game-backend/src/server.ts

```diff
@@ -7,9 +7,15 @@ const server = http.createServer();
 const WSServer = new WebSocket.Server({
   server,
 });
+const mainController = new MainController();
 
-WSServer.on('connection', ws => {
-  MainController.newConnection(ws);
+WSServer.on('connection', (ws: WebSocket, req: http.IncomingMessage) => {
+  const connectionId = req.headers['sec-websocket-key'] as string;
```

Comment:
Maybe we could extract this to a helper method `obtainConnectionId(request)` which returns an aliased type `ConnectionId` or `UserContextId` (similiar to what we're doing with `UserContextId` or `ObjectId`.